benchmarks/cugraph/pytest-based/bench_cugraph_uniform_neighbor_sample.py

```diff
@@ -169,44 +176,32 @@ def get_uniform_neighbor_sample_args(
     if with_replacement not in [True, False]:
         raise ValueError(f"got unexpected value {with_replacement=}")
 
-    rng = np.random.default_rng(seed)
-    num_verts = G.number_of_vertices()
-
-    if batch_size > num_verts:
-        num_start_verts = int(num_verts * 0.25)
-    else:
-        num_start_verts = batch_size
-
-    # Create the list of starting vertices by picking num_start_verts random
-    # ints between 0 and num_verts, then map those to actual vertex IDs.  Since
-    # the randomly-chosen IDs may not map to actual IDs, keep trying until
-    # num_start_verts have been picked, or max_tries is reached.
-    assert G.renumbered
-    start_list_set = set()
-    max_tries = 10000
-    try_num = 0
-    while (len(start_list_set) < num_start_verts) and (try_num < max_tries):
-        internal_vertex_ids_start_list = rng.choice(
-            num_verts, size=num_start_verts, replace=False
-        )
-        start_list_df = cudf.DataFrame({"vid": internal_vertex_ids_start_list})
-        start_list_df = G.unrenumber(start_list_df, "vid")
-
-        if G.is_multi_gpu():
-            start_list_series = start_list_df.compute()["vid"]
-        else:
-            start_list_series = start_list_df["vid"]
-
-        start_list_series.dropna(inplace=True)
-        start_list_set.update(set(start_list_series.values_host.tolist()))
-        try_num += 1
+    # Get the number of starting vertices (batch size) based on the graph size
+    # in order to support small graphs.  However, current benchmarks will never
+    # have graphs smaller than the batch size, so this is commented out.
+    # Uncomment if smaller graphs are being used.
+    #
+    # num_verts = G.number_of_vertices()
+    # if batch_size > num_verts:
+    #     num_start_verts = int(num_verts * 0.25)
+    # else:
+    #     num_start_verts = batch_size
+    num_start_verts = batch_size
+
+    # Create the list of starting vertices by simply picking the first
+    # num_start_verts vertices in the edgelist. This will likely result in
+    # duplicates.
+    srcs = G.edgelist.edgelist_df["src"]
+    start_list = srcs.head(num_start_verts)
```

Comment:
This might be a bit biased depending on how rmat generates the graph.  It might be better to renumber the graph and use `cupy.random.randint` to get a randomized start list.

Reply:
Changed this to get `num_start_verts` samples from the `src` Series in commit [0d60d77](https://github.com/rapidsai/cugraph/pull/3087/commits/0d60d771cc992dbabcd23088ec829916c84cf69a)